### PR TITLE
Add available_section to DocumentStructure

### PR DESCRIPTION
**Pull request by kyleknap**
List the sections currently in DocumentStructure. Note it does not list the subsections of its sections.

This will be needed for my impending botocore doc PR.

cc @jamesls @mtdowling

### CODE REVIEW

File: bcdoc/restdoc.py [hunk 1]
```diff
@@ -138,6 +138,10 @@ def path(self):
     def path(self, value):
         self._path = value
 
+    @property
+    def available_sections(self):
+        return list(self._structure.keys())
```
Comment (jamesls) [inline bcdoc/restdoc.py]: If _structure is a dict you don't need the `.keys()` call.

**Comment by jamesls** [comment]
:shipit: Just a small comment.

File: bcdoc/restdoc.py (hunk 1 above)
Comment (kyleknap) [inline bcdoc/restdoc.py]: In python3, I was getting a view object back that had some deviations from list behavior. So I just forced to be a list to be consistent across python versions.

Comment (jamesls) [inline bcdoc/restdoc.py]: Right, I'm saying you don't need the .keys() call.  Calling list on a dict will give you the keys.

**Comment by kyleknap** [comment]
I updated the PR based on feedback. Merging.

**Pull request merged**